kernel/sched.c

```diff
@@ -405,6 +405,8 @@ static void move_thread_to_end_of_prio_q(struct k_thread *thread)
 
 static int slice_ticks;
 static int slice_max_prio;
+struct _timeout slice_timeouts[CONFIG_MP_MAX_NUM_CPUS];
+bool slice_expired[CONFIG_MP_MAX_NUM_CPUS];
```

Comment:
No particular reason these couldn't go into the _current_cpu struct, though "struct _timeout" has a header ordering issue that would need to be fixed.  They're here just because this is where we have the timeslicing data right now.

Reply:
Another future improvement might be to try to unify this struct _timeout with the one already in the thread struct.  That was a little hairy, as there are likely lots of places that assume the thread timeout will only ever be used for suspended threads.  But it should definitely be possible: needless to say a thread can never be simultaneously running and timeslicing and suspended.